File: Sample/MaaSAdvertising-iOS-Sample/LandingPageViewController.m
//
//  LandingPageViewController.m
//  PWAdvertising
//
//  Created on 5/25/16.
//  Copyright © 2016 Phunware, Inc. All rights reserved.
//

#import "LandingPageViewController.h"
#import <PWAdvertising/PWAdsLandingPage.h>
#import <PWAdvertising/PWAdsRequest.h>

// Example zone id, don't use this in your app!
static NSString * const kZoneID =  @"76663";

@interface LandingPageViewController () <PWAdsLandingPageDelegate>

@end

@implementation LandingPageViewController {
    PWAdsLandingPage *_landingPage;
}

- (void)viewDidLoad {
    [super viewDidLoad];
    
    // Create PWAdsLandingPage instance
    _landingPage = [PWAdsLandingPage new];
    _landingPage.delegate = self;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)loadButtonPressed:(id)sender {
    // Create ad request with the given zone id.
    PWAdsRequest *adsRequest = [PWAdsRequest requestWithZoneID:kZoneID];
    
    // Set test mode for request. This will fetch test ads.
    adsRequest.testMode = YES;
    
    // Load the ad created request from the LandingPage view.
    [_landingPage loadAdsRequest:adsRequest];
}

#pragma mark -  PWAdsLandingPageDelegate

//  Called when a new advertisement is loaded.
- (void)landingPageDidLoadAd:(PWAdsLandingPage *)LandingPageAd {
    NSLog(@"landingPageDidLoadAd:");
    
    // LandingPage ad is loaded. Now present the LandingPage.
    [_landingPage presentFromViewController:self];
}

//  Called when an LandingPage fails to load advertisement.
- (void)landingPage:(PWAdsLandingPage *)landingPage didFailWithError:(NSError *)error {
    NSLog(@"landingPage:didFailWithError:");
}

// Called before LandingPage ad is dissmissed.
- (void)landingPageWillDismissModal:(PWAdsLandingPage *)landingPageAd {
    NSLog(@"landingPageWillDismissModal:");
}

// Called after landingPage ad has been dismissed.
- (void)landingPageDidDismissModal:(PWAdsLandingPage *)landingPageAd {
    NSLog(@"landingPageDidDismissModal:");
}

// Called before user leaves the application. This happens when user taps on an advertisment.
-(BOOL)shouldLeaveApplicationForlandingPage:(PWAdsLandingPage *)landingPage {
    NSLog(@"shouldLeaveApplicationForLandingPage:");
    return YES;
}

@end

File: Samples/Sample/Sample/LandingPageViewController.swift
//
//  LandingPageViewController.swift
//  Sample
//
//  Created on 8/15/17.
//  Copyright © 2017 Phunware, Inc. All rights reserved.
//

import UIKit
import PWAdvertising

struct LandingPageConstants {
    static let iPhoneZoneId = "76663"
    static let iPadZoneId = "76663"
}

class LandingPageViewController: UIViewController {
    
    // MARK: - Properties
    var zoneId: String!
    var request = PWAdsRequest()
    var landingPage = PWAdsLandingPage()

    // MARK: - IBActions
    @IBAction func load(_ sender: UIButton) {
        request.testMode = true
        print("Zone Id: \(request.zoneID!)\nTest mode is \(request.testMode)\n")
        landingPage.load(request)
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        landingPage.delegate = self
        
        switch UIDevice.current.userInterfaceIdiom {
        case .pad:
            zoneId = LandingPageConstants.iPadZoneId
        default:
            zoneId = LandingPageConstants.iPhoneZoneId
        }
        request.zoneID = zoneId
    }
}

extension LandingPageViewController: PWAdsLandingPageDelegate {
    func landingPageDidLoadAd(_ landingPageAd: PWAdsLandingPage!) {
        print("landingPageDidLoadAd\n")
        landingPage.present(from: self)
    }
    
    func landingPageDidDismissModal(_ landingPageAd: PWAdsLandingPage!) {
        print("landingPageDidDismissModal\n")
    }
    
    func landingPageDidPresentModal(_ landingPageAd: PWAdsLandingPage!) {
        print("landingPageDidPresentModal\n")
    }
    
    func landingPageWillDismissModal(_ landingPageAd: PWAdsLandingPage!) {
        print("landingPageWillDismissModal\n")
    }
    
    func landingPage(_ landingPage: PWAdsLandingPage!, didFailWithError error: Error!) {
        print("landingPage didFailWithError\n")
        if let error = error {
            print("\(error)\n")
        }
    }
    
    func shouldLeaveApplicationForlandingPage(_ landingPage: PWAdsLandingPage!) -> Bool {
        print("shouldLeaveApplicationForlandingPage\n")
        return true
    }
}
